viewing_party/party.py

```diff
@@ -1,23 +1,143 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+    return {"title": title, "genre": genre, "rating": rating}
 
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+            add_to_watched(user_data, movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_sum = 0
+    counter = 0
+    try:
+        for movie in user_data["watched"]:
+            rating_sum += movie["rating"]
+            counter += 1
+        return rating_sum / counter
+    except ZeroDivisionError:

```

Comment:
Good use of a `try/except`! I think it might be possible to avoid the possibility of a `ZeroDivisionError` if you use the guard clause pattern here. Something like:
```python
if not user_data["watched"]:
    return 0.0 # I think the requirements say to return a float if the list is empty, not an integer
# now you can assign your variables and do the `for` loop
```